### DISCUSSION TITLE
Title: Cannot Find Crazyradio Dongle in Oracle VM VirtualBox

### DISCUSSION THREAD

**Discussion by Iman-Kaviani123** [bitcraze/discussions#1427] [Q&A]
Hi everyone,

I've recently started working with Crazyflie in Oracle VM VirtualBox and encountered an issue that I need help with.

Setup:

I am using Oracle VM VirtualBox.
I downloaded the Crazyflie virtual machine (Crazyflie VM) from 
(https://github.com/bitcraze/bitcraze-vm/releases).
I am using the latest version (2023.10).
Issue:
When I try to run the Crazyflie software in the terminal, I see the following message:

`WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle
Cannot find a Crazyradio Dongle
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle
Cannot find a Crazyradio Dongle
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle
Cannot find a Crazyradio Dongle
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.`

as you can see many time i have been trying to click csan but Radio are not founding?!
you can see it in picture
<img width="372" alt="Unbenannt" src="https://github.com/bitcraze/discussions/assets/134901876/aea42140-b29e-4a79-b7b3-ab1f04662c6e">
<img width="920" alt="Unbenannt-1" src="https://github.com/bitcraze/discussions/assets/134901876/60dff894-d198-4c8e-a8af-74eaa90e05eb">

Any help or suggestions would be greatly appreciated!
 
Thanks in advance!
Iman

**Comment by ataffanel** [comment]
Are you passing the radio to the virtual machine using Virtualbox's menu?

One way to see if the radio is passed to the VM is to type the following in  a console in linux:
``` bash
$ lsusb -d 1915:7777
Bus 009 Device 016: ID 1915:7777 Nordic Semiconductor ASA Bitcraze Crazyradio (PA) dongle
```
If no Crazyradio is found, then you need to pass the radio to the VM.

**Comment by JonPrevost** [comment]
If you're like me, and have a brushless 2.1 that came with the Crazyradio 2.0, then you're SOL (stuck and out of luck). Get yourself a older Crazyradio PA, then pass that through. Run the updates, and now you should be able to find the radio and connect to your drone... except if it's the brushless model. I can ARM, and see the motors move to 10%, but nothing else works. Takeoff, land, up down, you name it, none of it works, so I think the VM is just too old and should be considered abandoned. I spent too much time working on it and think I'll just make my own VM and copy some of the bits from bitcraze's scripts.
Good luck. I know how frustrating this can be when updates brake basic functions. I feel like I'm being tortured for just trying lol
p.s. - The Android app can connect to the brushless... but it doesn't work, same problems. I regret getting this brushless model. Promised more payload, got more headaches.

**Reply by ataffanel** [reply]
The VM is indeed very old and we have mostly removed it from the documentation. The best way to work with Crazyflie is using a native Linux machine. Windows and Mac also tend to work with different amount of friction.

The VM has been challenging to maintain and we are still looking for ways to make it work better. In the meantime though, running native is the best way to work with Crazyflie.
